libs/api/ebpf_api.cpp

```diff
@@ -377,6 +407,56 @@ _get_program_information_data(ebpf_program_type_t program_type, ebpf_extension_d
     return retval;
 }
 
+uint32_t
+ebpf_get_program_byte_code(
+    const char* file_name,
+    const char* section_name,
+    ebpf_program_type_t* program_type,
+    bool mock_map_fd,
+    uint8_t** instructions,
+    uint32_t* instructions_size,
+    EbpfMapDescriptor** map_descriptors,
+    int* map_descriptors_count,
+    const char** error_message)
+{
+    ebpf_code_buffer_t byte_code(MAX_CODE_SIZE);
+    size_t byte_code_size = byte_code.size();
+    uint32_t result = ERROR_SUCCESS;
+
+    _map_file_descriptors.resize(0);
+
+    if (load_byte_code(
+            file_name, section_name, mock_map_fd, byte_code.data(), &byte_code_size, program_type, error_message) !=
+        0) {
+        result = ERROR_INVALID_PARAMETER;
+        goto Done;
+    }
+
+    // Copy instructions to output buffer.
```

Comment:
nit: It's good to end comments with punctuation, but this file should at least be consistent (see line 444 for instance)